python/lsst/pipe/tasks/colorterms.py

```diff
@@ -19,130 +19,157 @@
+        keytype=str,
+        itemtype=Colorterm,
+        default={},
+    )
 
-        @warning reverse is not yet implemented
-        """
-        if not colorterms:
-            colorterms = Colorterm._activeColorterms
+    # the following is desired to allow positional data and avoid DM-2382, but is impossible due to DM-2381
+    # def __init__(self, dict=None):
+    #     """!Construct a ColortermDictConfig
 
-        if not colorterms:
-            return primary
+    #     Overrides the default constructor for improved safety
+    #     and to allow the data to be a positional argument
+    #     """
+    #     if dict is None: # mutable objects should not be used as defaults
+    #         dict = {}
+    #     Config.__init__(self, dict=dict)
 
-        ct = colorterms[band]
-        
-        p = primary
-        s = secondary
 
-        if reverse:
-            raise NotImplemented("reverse photometric transformations are not implemented")
-        else:
-            return p + ct.c0 + (p - s)*(ct.c1 + (p - s)*ct.c2)
+class ColortermLibraryConfig(Config):
+    """!A config containing library: a dict of catalog name: ColortermDictConfig
 
-    @staticmethod
-    def propagateFluxErrors(band, primaryFluxErr, secondaryFluxErr, reverse=False, colorterms=None):
-        """!Transform flux errors
+    This is intended to support a particular camera with a variety of reference catalogs
+    """
+    library = ConfigDictField(
+        doc="Mapping of reference catalog name (or glob) to group of color terms",
+        keytype=str,
+        itemtype=ColortermDictConfig,
+        default={},
+    )
+
+    # the following is desired to allow positional data and avoid DM-2382, but is impossible due to DM-2381
+    # def __init__(self, library=None):
+    #     """!Construct a ColortermLibraryConfig
+
+    #     Overrides the default constructor for improved safety
+    #     and to allow the data to be a positional argument
+    #     """
+    #     if library is None: # mutable objects should not be used as defaults
+    #         library = {}
+    #     Config.__init__(self, library=library)
+
+    def getColorterm(self, filterName, refCatName, doRaise=True):
+        """!Get the appropriate Colorterm from the library
+
+        We use the group of color terms in the library that matches the refCatName.
+        If the refCatName exactly matches an entry in the library, that
+        group is used; otherwise if the refCatName matches a single glob (shell syntax,
+        e.g., "sdss-*" will match "sdss-dr8"), then that is used.  If there is no
+        exact match and no unique match to the globs, we raise an exception.
+
+        @param filterName  name of filter
+        @param refCatName  reference catalog name or glob expression; if a glob expression then
+            there must be exactly one match in the library
+        @param[in] doRaise  if True then raise ColortermNotFoundError if no suitable Colorterm found;
```

Comment:
I fixed it so expected errors are all ColortermNotFoundError and raising them is controlled by doRaise. Unexpected errors will break through.